.github/workflows/ci-low-cadence.yml

```diff
@@ -20,6 +20,11 @@ jobs:
         java: [ '8', '11', '16', '17-ea' ]
         os: [ 'ubuntu-20.04', 'windows-latest' ]
     steps:
+      - name: Setup java
```

Comment:
Caching jobs are not executing Gradle scripts. So it's ok to leave `setup-java` where it was. Also in this particular case we call `setup-java` twice:
- First time to install the target JVM (e.g. 17-ea).
- Second time to install the JDK 8 with which we are going to execute the Gradle script. 

The two are different, because Gradle is not able to run on JDK 17 yet hence we must run the script with a version that always works but then specify the target JVM we want for our build via the `BUILD_JAVA_HOME` and `BUILD_JAVA_VERSION` environment variables.

Reply:
I'll revert that change out and go with the one that you've already made to avoid the auto-detect.